src/ethereum_test_tools/vm/opcode.py

```diff
@@ -205,171 +205,4898 @@ class Opcodes(Opcode, Enum):
+PUSH28() = value
+----
+
+Description
+----
+Place 28 byte item on stack
+
+Inputs
+----
+- None
+
+Outputs
+----
+- value: pushed value, aligned to the right (put in the lowest significant bytes)
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#7B](https://www.evm.codes/#7B)
+"""
+
+PUSH29 = Opcode(0x8C, pushed_stack_items=1, data_portion_length=29)
+"""
+PUSH29() = value
+----
+
+Description
+----
+Place 29 byte item on stack
+
+Inputs
+----
+- None
+
+Outputs
+----
+- value: pushed value, aligned to the right (put in the lowest significant bytes)
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#7C](https://www.evm.codes/#7C)
+"""
+
+PUSH30 = Opcode(0x8D, pushed_stack_items=1, data_portion_length=30)
+"""
+PUSH30() = value
+----
+
+Description
+----
+Place 30 byte item on stack
+
+Inputs
+----
+- None
+
+Outputs
+----
+- value: pushed value, aligned to the right (put in the lowest significant bytes)
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#7D](https://www.evm.codes/#7D)
+"""
+
+PUSH31 = Opcode(0x8E, pushed_stack_items=1, data_portion_length=31)
+"""
+PUSH31() = value
+----
+
+Description
+----
+Place 31 byte item on stack
+
+Inputs
+----
+- None
+
+Outputs
+----
+- value: pushed value.
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#7E](https://www.evm.codes/#7E)
+"""
+
+PUSH32 = Opcode(0x8F, pushed_stack_items=1, data_portion_length=32)
+"""
+PUSH32() = value
+----
+
+Description
+----
+Place 32 byte item on stack
+
+Inputs
+----
+- None
+
+Outputs
+----
+- value: pushed value.
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#7F](https://www.evm.codes/#7F)
+"""
+
+DUP1 = Opcode(0x80, pushed_stack_items=1, min_stack_height=1)
+"""
+DUP1(value) = value
+----
+
+Description
+----
+Duplicate 1st stack item
+
+Inputs
+----
+- value: value to duplicate
+
+Outputs
+----
+- value: duplicated value.
+- value: original value
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#80](https://www.evm.codes/#80)
+"""
+
+DUP2 = Opcode(0x81, pushed_stack_items=1, min_stack_height=2)
+"""
+DUP2(a, b) = b,a,b
+----
+
+Description
+----
+Duplicate 2nd stack item
+
+Inputs
+----
+- a: ignored value.
+- b: value to duplicate.
+
+Outputs
+----
+- b: duplicated value.
+- a: ignored value.
+- b: original value.
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#81](https://www.evm.codes/#81)
+"""
+
+DUP3 = Opcode(0x82, pushed_stack_items=1, min_stack_height=3)
+"""
+DUP15(a, b, c) = c, a, b, c
+----
+
+Description
+----
+Duplicate 3rd stack item
+
+Inputs
+----
+- a: ignored value.
+- b: ignored value.
+- c: value to duplicate.
+
+Outputs
+----
+- c: duplicated value.
+- a: ignored value.
+- b: ignored value.
+- c: original value.
+
+Fork
+----
+Frontier
+
+Gas
+----
+3
+
+Source: [evm.codes/#82](https://www.evm.codes/#82)
+"""
+
+DUP4 = Opcode(0x83, pushed_stack_items=1, min_stack_height=4)
+"""
+DUP15(value1, value2, value3, value) = value, value2, value3, value
```

Comment:
```suggestion
DUP4(value1, value2, value3, value) = value, value1, value2, value3, value
```
Similar issue on the rest of the DUP opcodes.